### DISCUSSION TITLE
Waterlogged blocks

### DISCUSSION THREAD

**Discussion by Kerrghan** [Roadhog360/Et-Futurum-Requiem#106] [Ideas]
It would be really nice to be able to waterlog at least slabs and stairs. I really love to use waterlogged slabs and stairs for decoration (and for not falling into the water of my farm lol), sadly can't do so if I'm playing a mod that stayed in 1.7.10...

**Comment by GregoriusT** [comment]
I do not know how well waterlogging Blocks would work in 1.7.10, but what I do know is that you could put a Lily Pad on the Water Sources of your Farms.

Now that I think of it, Water requires "Material.water" to be used in order to attach to it (what is a major annoyance if you are a Mod that adds other Fluid Blocks...), which is fine if you are okay with waterlogged Slabs and Stairs being a different Material Type than Stone or Wood. So maybe it "could" work, but it would require a lot of overhead to make it work properly.

**Reply by Kerrghan** [reply]
Wauw, can't believe I forgot the existence of lily pads... especially since they are so annoying with boats in 1.7... thx for the reminder

**Comment by Roadhog360** [comment]
Removed "Won't Add" label as this is now in consideration.

**Comment by GregoriusT** [comment]
Just so you know, using Material.water or Material.lava in any way will screw you over, because Mods love to use it to declare that something is a full Fluid Block. Would be kinda hilarious if you have a shipwreck and could pump away the waterlogged Blocks from it. XD

**Reply by Roadhog360** [reply]
I know, it'll just be the regular block but water rendered inside of it and water physics added.

**Reply by GregoriusT** [reply]
good luck with the water physics that dont use Material.water because holy crap the Mojang Code is THAT bad...

**Reply by Roadhog360** [reply]
I am just going to mixin entity.class to run the water physics and is in water check if you are swimming in a waterlogged block.

**Comment by Kerrghan** [comment]
Huh, didn't think I would see an update on this 1 year and a half later haha, happy to see this feature will be added, good luck!

**Reply by Roadhog360** [reply]
No, it's not a promise, it's merely just being considered. Plans to look into it are being made.

**Reply by Kerrghan** [reply]
Yes sorry my bad, bad wording, I did understand it was just a consideration, guess I got too excited haha

**Comment by DarianLStephens** [comment]
If you haven't already, you could peek at [Fluidlogged API on Modrinth](https://modrinth.com/mod/fluidlogged-api), adding a waterlogging system to 1.12.

**Reply by Roadhog360** [reply]
It's probably going to be a useless reference because it's likely just going to use BlockStates in the same or similar way that 1.13 does. In these versions storing basic data in non-tile entities is a solved problem.

**Reply by DarianLStephens** [reply]
It has separate chunk data. From what I can tell, it doesn't use block states at all, especially because it can support any block and fluid.

**Reply by Roadhog360** [reply]
That's odd, because block states were actually added in 1.8 and available to use from then on, up to and including 1.12, then 1.13 removes meta and makes BlockStates the only way for variated blocks. Does this mod really use new chunk data instead of BlockStates?

**Reply by DarianLStephens** [reply]
It says as much in the description, and the source code matches up, using chunk capabilities for its new data.
You could maybe use [FalsePattern's ChunkAPI](https://modrinth.com/mod/chunkapi) as a substitute for capabilities, if you don't want to write a bunch of that stuff yourself.

**Comment by Roadhog360** [comment]
Unfortunately I have permanently decided to re-add this to the list of ideas not considered, as fluid logging would best be handled by another mod. This feature is also far above my skill level. But yeah, fluid logging is best handled by another mod due to the chunk data required to add, and modders might want to support modded fluids too
